src/onewire.rs

```diff
@@ -0,0 +1,102 @@
+use core::marker::PhantomData;
+use core::ptr;
+
+use esp_idf_sys::*;
+
+use crate::peripheral::Peripheral;
+
+#[derive(Debug)]
+pub struct Device<'b> {
+    _device: onewire::onewire_device_t,
+    // _bus: &'b BusDriver<'b>, // not sure how I would hold a reference to this aside from the handle
+    _p: PhantomData<&'b ()>, // holds the lifetime since the device is linked to the lifetime of the bus
+}
+
+impl<'a> Device<'a> {
+    fn new(device_handle: onewire::onewire_device_t) -> Self {
+        Self {
+            _device: device_handle,
+            _p: PhantomData,
+        }
+    }
+
+    /// get the handle of the bus the device is attached to
+    pub fn bus(&self) -> onewire::onewire_bus_handle_t {
+        self._device.bus
+    }
+
+    /// get the device address
+    pub fn address(&self) -> u64 {
+        self._device.address
+    }
+}
+
+/// wrapper around the device iterator to search for available devices on the bus
+pub struct DeviceSearch {
+    _search: onewire::onewire_device_iter_handle_t,
+}
+
+impl DeviceSearch {
+    fn new(bus: &mut BusDriver) -> Result<Self, EspError> {
+        let mut my_iter: onewire::onewire_device_iter_handle_t = ptr::null_mut();
+
+        esp!(unsafe { onewire::onewire_new_device_iter(bus._bus, &mut my_iter) })?;
+
+        Ok(Self { _search: my_iter })
+    }
+
+    pub fn next_device(&mut self) -> Result<Device, EspError> {
+        let mut next_onewire_device = onewire::onewire_device_t::default();
+        esp!(unsafe {
+            onewire::onewire_device_iter_get_next(self._search, &mut next_onewire_device)
+        })?;
+
+        Ok(Device::new(next_onewire_device))
+    }
+}
+
+impl Drop for DeviceSearch {
```

Comment:
I see this is done for other peripherals, do I need  to do it here as well?

Reply:
@ivmarkov

Reply:
Not sure what is the question? Do you need to do _what_ w.r.t. `DeviceSearch`??

Reply:
Do I need to impl drop in this manner, as other peripherals have done?